src/cache/cache_las.c

```diff
@@ -829,26 +830,23 @@ __wt_las_cursor_position(WT_CURSOR *cursor, uint64_t pageid)
 		cursor->set_key(cursor,
 		    pageid, (uint32_t)0, (uint64_t)0, &las_key);
 		WT_RET(cursor->search_near(cursor, &exact));
-		if (exact < 0) {
+		if (exact < 0)
 			WT_RET(cursor->next(cursor));
 
-			/*
-			 * Because of the special visibility rules for
-			 * lookaside, a new block can appear in between our
-			 * search and the block of interest.  Keep trying while
-			 * we have a key lower than we expect.
-			 *
-			 * There may be no block of lookaside entries if they
-			 * have been removed by
-			 * WT_CONNECTION::rollback_to_stable.
-			 */
-			WT_RET(cursor->get_key(cursor,
-			    &las_pageid, &las_id, &las_counter, &las_key));
-			if (las_pageid < pageid)
-				continue;
-		}
-
-		return (0);
+		/*
+		 * Because of the special visibility rules for
+		 * lookaside, a new block can appear in between our
+		 * search and the block of interest.  Keep trying while
+		 * we have a key lower than we expect.
+		 *
+		 * There may be no block of lookaside entries if they
+		 * have been removed by
+		 * WT_CONNECTION::rollback_to_stable.
+		 */
+		WT_RET(cursor->get_key(cursor,
+		    &las_pageid, &las_id, &las_counter, &las_key));
+		if (las_pageid >= pageid)
+			return (0);
```

Comment:
This is just paranoia: no need to keep it.

Reply:
@agorrod, should this be reverted before the merge? I don't think it's a necessary change, and it potentially changes the performance of `__wt_las_cursor_position()`.

(I don't know if that matters or not.)

Reply:
I figured that since @michaelcahill was suspicious of the previous behavior, and it's a reasonable change we might as well keep it. We are in the cache overflow path - so I'm not really concerned that doing a get_key will noticeably change performance.